bin/modules/birthday/birthday_module.dart

```diff
@@ -0,0 +1,134 @@
+import 'dart:async';
+
+import 'package:cron/cron.dart';
+import 'package:hive/hive.dart';
+import 'package:nyxx/nyxx.dart';
+import 'package:nyxx_interactions/nyxx_interactions.dart';
+
+import '../bot_module.dart';
+import 'hive_constants.dart';
+
+part 'commands.dart';
+
+class BirthdayModule extends BotModule {
+  static final BirthdayModule _singleton = BirthdayModule._internal();
+
+  bool _isInitialized = false;
+
+  late INyxxWebsocket _bot;
+
+  final Cron _birthdayCron = Cron();
+
+  factory BirthdayModule() {
+    return _singleton;
+  }
+  BirthdayModule._internal();
+
+  @override
+  void init(INyxxWebsocket bot) {
+    _bot = bot;
+
+    // Set up Hive for local data storage
+    Hive.init('/data');
+    Hive.openBox(HiveConstants.birthdayBox);
+
+    _birthdayCron.schedule(Schedule.parse('0 9 * * *'), () {
+      _checkBirthdays();
+    });
+
+    _isInitialized = true;
+  }
+
+  void _checkBirthdays() {
+    if (!_isInitialized) {
+      print('Birthday module not initialized!');
+      throw Exception('Birthday module not initialized!');
+    }
+    var box = Hive.box(HiveConstants.birthdayBox);
+
+    Map<dynamic, dynamic> listOfUsers =
+        box.get(HiveConstants.birthdayUserList, defaultValue: {});
+
+    var updateChannelId = box.get(HiveConstants.birthdayChannelId);
+
+    if (updateChannelId == null) {
+      print('No channel available for updates!');
+      return;
+    }
+
+    if (listOfUsers.isEmpty) print('No users available to scrape!');
+
+    listOfUsers.forEach((userSnowflake, birthday) async {
+      var birthdayDate = DateTime.parse(birthday);
+      var birthdayUser = await _bot.fetchUser(Snowflake(userSnowflake));
+
+      // If a user has birthday today, post update to main update channel!
+      if (birthdayDate.day == DateTime.now().day &&
+          birthdayDate.month == DateTime.now().month) {
+        var channel =
+            await _bot.fetchChannel(Snowflake(updateChannelId)) as ITextChannel;
+
+        await channel.sendMessage(MessageBuilder.content(
+            'Happy birthday ${birthdayUser.mention}! 🎉🎉🎉'));
+      }
```

Comment:
If the "CheckBirthday function is only called once a day, this will be fine but if it checks several times won't it post "Happy birthday" several times?